app/controllers/NotificationsController.scala

```diff
@@ -38,18 +39,22 @@ class NotificationsController @Inject()(
     extends FrontendController(mcc) with I18nSupport {
 
   def listOfNotifications(conversationId: String): Action[AnyContent] = authenticate.async { implicit request =>
-    for {
-      submission <- connector.fetchSingleSubmission(conversationId)
-      submissionElem = submission.map(factory.build)
-
-      notificationsElems <- connector
-        .fetchNotifications(conversationId)
-        .map(notifications => notifications.sorted.map(factory.build))
-
-      submissionUcr = submission.flatMap(extractUcr).getOrElse("")
-      elementsToDisplay = (submissionElem.toSeq ++ notificationsElems)
-
-    } yield Ok(notifications(submissionUcr, elementsToDisplay))
+    val params = for {
```

Comment:
I think with map it'll be easier to read and shorter, but it's just my opinion, non blocking comment

Reply:
I tried various methods, using a map, using chained for comprehension. None seemed to compile :(